src/components/Signup/Signup.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { 
+  ChangeEvent, 
+  FormEvent, 
+  useState 
+} from "react";
+
+import "./Signup.css";
+
+const Signup = (): JSX.Element => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if( event.target.name == "email") {
+      setEmail( event.target.value );
+    } else {
+      setPassword( event.target.value );
+    }
```

Comment:
Since the `formState` is grouped together, you can now make the _change event_ more dynamic by setting the key from the `formState` object to the `name` property from `event.target`:
```tsx
setForm(prevState => ({
  ...prevState,
  [event.target.name]: event.target.value
});
```
So I would make those chages.